### PR TITLE
Add `index page` section to documentation

### PR DESCRIPTION
**Pull request by codesections**
The current documentation does not describe how to create a index page.  I initially found this confusing, because I expected an index page to be a **page** rather than a section.  Thus, I tried to access the page content with `{{ page.content }}` and was very frustrated when I could not.

This addition clarifies that the index page is **always** a section, even if it does not have any sub-pages.  This should also help people who intend to use Gutenberg to build stand-alone webpages, rather than blogs.

(Also, please let me know if these documentation related pull requests are unwelcome/annoying.  I'm really enjoying using Gutenberg, and I feel that the best way I can help right now—given that I don't (yet) know Rust well enough to contribute code—is by suggesting ways that the documentation could be clearer to address the spots I got stuck.  But if it's not helpful, let me know and I'll knock it off.)

### CODE REVIEW

**Comment by Keats** [comment]
I'm not sure this deserves a special page of its own. Can this be a small subection on https://www.getgutenberg.io/documentation/templates/overview/ before the built-in filter one? It is mentioned in https://www.getgutenberg.io/documentation/templates/pages-sections/#section-variables

I think it can be shorter, for example just saying that the index page is always a section and to refer to the page & section for more details.

> (Also, please let me know if these documentation related pull requests are unwelcome/annoying. I'm really enjoying using Gutenberg, and I feel that the best way I can help right now—given that I don't (yet) know Rust well enough to contribute code—is by suggesting ways that the documentation could be clearer to address the spots I got stuck. But if it's not helpful, let me know and I'll knock it off.)

Documentation updates are great, keep them coming!

**Comment by codesections** [comment]
I moved the info to the `templates/overview` section as you suggested.  I agree that it's a much better fit there.

I'm not sure I kept it quite as short as you were suggesting, though.  I thought it was helpful to add a bit more context, but let me know if you think that's too much and I can cut it down a bit.

File: docs/content/documentation/templates/overview.md [hunk 1]
```diff
@@ -23,6 +23,47 @@ A few variables are available on all templates minus RSS and sitemap:
...
+By default, Gutenberg will look for three templates: `index.html`, which is applied 
+to the site homepage; `section.html`, which is applied to all sections (any HTML 
+page generated by creating a directory within your `content` directory); and 
+`page.html`, which is applied to all pages (any HTML page generated by creating a 
+`.md` file within your `content` directory).
+
+The homepage is always a section (regardless of whether it contains other pages).
+Thus, the `index.html` and `section.html` templates both have access to the 
+section variables.  The `page.html` template has access to the page variables.
+This means that you can, for example, access the content of your `_index.md` files
```
Comment (Keats) [inline docs/content/documentation/templates/overview.md]: I would remove that sentence

File: docs/content/documentation/templates/overview.md [hunk 2]
```diff
@@ -23,6 +23,47 @@ A few variables are available on all templates minus RSS and sitemap:
...
+
+```md
++++
+title = "About Us"
+template = "about.html"
++++
+```
+
+Custom templates are not required to live at the root of your `templates` directory.
+For example, `product_pages\with_pictures.html` is a valid template.
```
Comment (Keats) [inline docs/content/documentation/templates/overview.md]: \ -> / to have slashes consistent

File: docs/content/documentation/templates/overview.md [hunk 3]
```diff
@@ -23,6 +23,47 @@ A few variables are available on all templates minus RSS and sitemap:
...
+
+The one time that custom templates _are_ applied without being specified in the front
```
Comment (Keats) [inline docs/content/documentation/templates/overview.md]: I would remove that paragraph, they each have a dedicated page in the docs

**Review by Keats — CHANGES_REQUESTED** [review]

**Comment by codesections** [comment]
I made the changes you requested.  I also moved a sentence from earlier in the page down to its own section to clarify the nature of built-in templates without adding (much) text.

**Comment by Keats** [comment]
Thanks!

**Pull request merged**